uber/tests/models/test_badge_funcs.py

```diff
@@ -15,10 +15,11 @@ def session(request):
 
 @pytest.fixture(autouse=True)
 def teardown_range_check(request):
-    def _check_range():
-        with Session() as session:
-            check_ranges(session)
-    request.addfinalizer(_check_range)
```

Comment:
Two things.

First, I think this should say ``if c.SHIFT_CUSTOM_BADGES`` rather than ``if not c.SHIFT_CUSTOM_BADGES``.  I think when I was rattling off how you could just do this as an if statement in my earlier comment, I accidentally reversed it.  Sorry about that.  But we WANT to check the ranges anytime shifting is turned on (because in that case everything should line up) but we DON'T want to check the range when shifting is turned off (since then we could e.g. have a hole in the middle).

2) This placement of the if-statement may not end up being correct in all cases; you should need to put the ``if`` inside the definition of ``_check_ranges()``.  The reason is that we could end up with the following sequence of things:
- pytest runs this fixture when ``c.SHIFT_CUSTOM_BADGES`` is ``True``, so it sets ``_check_ranges()`` as a finalizer
- a test monkeypatches ``c.SHIFT_CUSTOM_BADGES`` to ``False``
- the test completes, and we run the finalizer, because we set it to run before we monkeypatched the value

So I think this function should really read:

```python
@pytest.fixture(autouse=True)
def teardown_range_check(request):
    def _check_range():
        if c.SHIFT_CUSTOM_BADGES:
            with Session() as session:
                check_ranges(session)
        request.addfinalizer(_check_range)
```

Sorry for not spelling that our more clearly earlier.